### ISSUE TITLE
Allow authentication with an oauth bearer token

### ISSUE THREAD

**Issue by nojo** [forcedotcom/wsc#183]
API access via oauth is generally considered more secure than username/password, and Salesforce is encouraging oauth usage for APIs: https://developer.salesforce.com/page/Digging_Deeper_into_OAuth_2.0_on_Force.com.  Official client should support oauth.

**Comment by nojo** [comment]
Apparently a bearer token can be used anywhere that a sessionId is used.  So for cases where you have already acquired a bearer token, it's just a matter of using that as the sessionId.  Would still be handy to support acquiring a bearer token via the client rather than outside of it.

**Comment by reubencornel** [comment]
Yes we(internally) agree that we should be able to support retrieving the bearer token using the library. I've created a user story internally to support it, but its not been scheduled for a fix yet.
